app/cognito/utils/client.py

```diff
@@ -0,0 +1,53 @@
+from boto3 import client
+from mypy_boto3_cognito_idp.type_defs import AdminGetUserResponseTypeDef
+from mypy_boto3_cognito_idp.type_defs import UserTypeTypeDef
+
+from django.conf import settings
+
+
+class Client:
+    """ A low level client for managing cognito users. """
+
+    def __init__(self) -> None:
+        self.endpoint_url = settings.COGNITO_ENDPOINT_URL
+        self.user_pool_id = settings.COGNITO_POOL_ID
+        self.client = client("cognito-idp", endpoint_url=self.endpoint_url)
+
+    def get_users(self) -> list[UserTypeTypeDef]:
+        response = self.client.list_users(UserPoolId=self.user_pool_id)
+        return response['Users']
+
+    def get_user(self, username: str) -> AdminGetUserResponseTypeDef | None:
+        try:
+            response = self.client.admin_get_user(UserPoolId=self.user_pool_id, Username=username)
+        except self.client.exceptions.UserNotFoundException:
+            return None
+
+        return response
+
+    def create_user(self, username: str, email: str) -> UserTypeTypeDef:
+        response = self.client.admin_create_user(
+            UserPoolId=self.user_pool_id,
+            Username=username,
```

Comment:
Am I right that the username is the field that identifies the records uniquely? cc @asteiner-swisstopo

Reply:
I just saw that for the moment you chose `User.id` for the username. But I think we should have a designated field for that in the model.

Reply:
Agree, `user.id` is a bad choice, since it's implicit and automatically set. I would also prefer a designated field.